app/test/unit/git/stash-test.ts

```diff
@@ -61,47 +62,89 @@ describe('git/stash', () => {
     })
 
     it('creates a stash entry', async () => {
-      await await FSE.appendFile(readme, 'just testing stuff')
-      const tipSha = await getTipSha(repository)
+      await FSE.appendFile(readme, 'just testing stuff')
+      const { sha } = await getTipOrError(repository)
 
-      await createDesktopStashEntry(repository, 'master', tipSha)
+      await createDesktopStashEntry(repository, 'master', sha)
 
       const result = await GitProcess.exec(['stash', 'list'], repository.path)
       const entries = result.stdout.trim().split('\n')
       expect(entries).toHaveLength(1)
     })
   })
-})
 
-async function getTipSha(repository: Repository) {
-  const result = await GitProcess.exec(['rev-parse', 'HEAD'], repository.path)
-  return result.stdout.trim()
-}
+  describe('getLastDesktopStashEntry', () => {
+    let repository: Repository
+    let readme: string
+
+    beforeEach(async () => {
+      repository = await setupEmptyRepository()
+      readme = path.join(repository.path, 'README.md')
+      await FSE.writeFile(readme, '')
+      await GitProcess.exec(['add', 'README.md'], repository.path)
+      await GitProcess.exec(['commit', '-m', 'initial commit'], repository.path)
+    })
 
-async function stash(repository: Repository, message?: string) {
-  const tipSha = await getTipSha(repository)
+    it('returns null when no stash entries exist for branch', async () => {
```

Comment:
What about a test when no stashes are created in the repository?

Reply:
The function operates at the branch level, so I don't think testing for stashes in a repo would give us much. I've updated the name of the function to make its purpose more clear.

Reply:
I wasn't that worried about the branch name, given it's parameterized accordingly - I just didn't quite understand this test summary versus the implementation.

What about this test description for the current test:

>  `returns null when no stash entries match branch name`

This might then motivate creating an additional test to cover a related scenario

> `returns null when no stash entries created`

Reply:
> The function operates at the branch level, so I don't think testing for stashes in a repo would give us much. 

This might be correct, but we also don't have a test for `getDesktopStashEntries` to ensure it handles when no stashes exist. We only cover "unborn repository" and "returns created stashes". That's the gap I'd like to cover here.